docs/README.md

```diff
@@ -0,0 +1,113 @@
+# Scylla documentation
```

Comment:
I don't understand what's the difference between README.md and index.rst - are they meant to be shown in different places?

If the README.md is not shown on the Scylla site but is for ordinary users, maybe it should begin by explaining the different role of the "dev" subdirectory vs. everything else.

But we can do this later.

Reply:
The index.rst file is the source for the documentation landing page. 
README.md is not included in the published docs. It should include information about the docs relevant to the contributors (what is included, how to run preview, etc.).

I'm going to update both files later on because the landing page needs to be restructured and the README.md file should be clearer. I didn't want to do it as part of this PR as its purpose is to migrate the docs rather than improve them.